android/src/main/java/com/thunderhead/one_sdk_flutter/OnePlugin.java

```diff
@@ -29,132 +40,271 @@
+                final Map<String, Object> arguments = call.arguments();
+                String interactionPath = (String) arguments.get("interactionPath");
+                HashMap<String, Object> properties = (HashMap) arguments.get("properties");
+                try {
+                    sendInteraction(interactionPath, properties, result);
+                } catch (Exception e) {
+                    result.error("Failed to send interaction", e.getLocalizedMessage(), e);
+                }
+            } else if (call.method.equals("sendResponseCode")) {
+                final Map<String, String> arguments = call.arguments();
+                String responseCode = (String) arguments.get("responseCode");
+                String interactionPath = (String) arguments.get("interactionPath");
+                sendResponseCode(responseCode, interactionPath, result);
+            } else if (call.method.equals("setLogLevel")) {
+                setLogLevel(call, result);
+            } else if (call.method.equals("initializeOne")) {
+                initializeOne(call, result);
+            } else if (call.method.equals("optOut")) {
+                optOut(call, result);
+            } else {
+                result.notImplemented();
+            }
+        } catch (Exception error) {
+            Log.e(LOG_TAG, "[Thunderhead] Method Call Exception Error: " + error.getLocalizedMessage());
+            result.error(Integer.toString(error.hashCode()), error.getLocalizedMessage(), error);
+        }
+    }
+
+    private void initializeOne(MethodCall call, final Result result) {
+        final Map<String, Object> arguments = call.arguments();
+        String siteKey = (String) arguments.get("siteKey");
+        String touchpointURI = (String) arguments.get("touchpoint");
+        String apiKey = (String) arguments.get("apiKey");
+        String sharedSecret = (String) arguments.get("sharedSecret");
+        String userId = (String) arguments.get("userID");
+        String host = (String) arguments.get("host");
+        boolean adminMode = (Boolean) arguments.get("adminMode");
+
+        final OneConfiguration oneConfiguration = new OneConfiguration.Builder()
+                .siteKey(siteKey)
+                .apiKey(apiKey)
+                .sharedSecret(sharedSecret)
+                .userId(userId)
+                .host(URI.create(host))
+                .touchpoint(URI.create(touchpointURI))
+                .mode(adminMode ? OneMode.ADMIN : OneMode.USER)
+                .build();
+        One.setConfiguration(oneConfiguration);
+        result.success(null);
+    }
+
+    private void setLogLevel(MethodCall call, final Result result) {
+        OneLoggingConfiguration oneLoggingConfiguration;
+        final Map<String, Boolean> arguments = call.arguments();
+        Boolean enabled = arguments.get("logLevel");
+
+        if (enabled == null) {
+            enabled = true;
+        }
+
+        if (enabled) {
```

Comment:
When we switched to the builder pattern for java we envisioned use like this (reduces boilerplate):

```
OneLoggingConfiguration.Builder builder = OneLoggingConfiguration.builder()
    // common to either path
    .log(OneLogComponent.Any);
if (enabled) {
    builder.log(OneLogLevel.VERBOSE).log(OneLogLevel.DEBUG)
} else {
    builder.log(OneLogLevel.WARN).log(OneLogLevel.ERROR)
}
One.setLoggingConfiguration(builder.build());
```